components/userDashboard/Bookings.js

```diff
@@ -27,66 +24,72 @@ const Bookings = () => {
       );
       setSessions(response.data);
     } catch (error) {
-      setError(error.response.data.message);
+      setError(error.response?.data?.message || "An error occurred");
     }
   };
 
-  // calling fetch function onload
   useEffect(() => {
     fetchSession();
   }, []);
 
+  const formatDate = (isoDate) => {
+    if (!isoDate) return "...";
+    const date = new Date(isoDate);
+    const day = date.getDate();
+    const month = date.toLocaleString("default", { month: "short" });
+    const year = date.getFullYear();
+    return `${day} ${month} ${year}`;
+  };
+
+  const filterFunction = (session, activeTab) => {
+    return (
+      (activeTab.toLowerCase() === "pending" && session.isbooked) ||
+      (activeTab.toLowerCase() === "completed" && !session.isbooked)
+    );
+  };
+
+  const sessionData = sessions.map((session) => [
+    session.sessionName,
+    session.mentorName,
+    session.sessionDate,
+    session.sessionTime,
+  ]);
+
   return (
     <div className="tw-p-3 md:tw-p-8 tw-w-full">
-      <h1 className="tw-text-2xl sm:tw-text-3xl lg:tw-text-4xl tw-font-semibold tw-mb-4  lg:tw-mb-8">
+      <h1 className="tw-text-2xl sm:tw-text-3xl lg:tw-text-4xl tw-font-semibold tw-mb-4 lg:tw-mb-8">
         Your sessions
       </h1>
 
-      {/* Find mentors */}
       <div className="tw-flex tw-items-center tw-gap-4 tw-mb-8">
         <p className="tw-text-base-500 tw-font-semibold">Find mentors</p>
         <Link href="/mentorList">
-          <button className="tw-bg-primary-100 tw-text-white tw-p-2 tw-rounded-md tw-font-semibold hover:tw-bg-primary-200 tw-duration-150 tw-ease-in-out">
-            here
-          </button>
+          {!loader ? (
+            <button
+              onClick={() => setLoader(true)}
+              className="tw-bg-primary-100 tw-text-white tw-p-2 tw-rounded-md tw-font-semibold hover:tw-bg-primary-200 tw-duration-150 tw-ease-in-out"
+            >
+              here
+            </button>
+          ) : (
+            <Loader width="25px" />
+          )}
         </Link>
       </div>
```

Comment:
we we are using here `onclick` can we include redirecting in it ? we can avoid then link